test/e2e/data/cluster-templates/docker-rke2.yaml

```
@@ -0,0 +1,168 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: DockerCluster
+metadata:
+  name: ${CLUSTER_NAME}
+  annotations:
+    "helm.sh/resource-policy": keep
+spec:
+  loadBalancer:
+    customHAProxyConfigTemplateRef:
+      name: ${CLUSTER_NAME}-lb-config
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster 
+metadata:
+  name: ${CLUSTER_NAME} 
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 10.45.0.0/16
+    services:
+      cidrBlocks:
+      - 10.46.0.0/16
+    serviceDomain: cluster.local
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: RKE2ControlPlane
+    name: ${CLUSTER_NAME}-control-plane
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: DockerCluster
+    name: ${CLUSTER_NAME}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: DockerMachineTemplate
+metadata:
+  name:  ${CLUSTER_NAME}-control-plane
+spec:
+  template:
+    spec:
+      customImage: kindest/node:v1.28.0
+      bootstrapTimeout: 15m
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha1
+kind: RKE2ControlPlane
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+spec: 
+  replicas: ${CONTROL_PLANE_MACHINE_COUNT}
+  agentConfig:
+    version: ${KUBERNETES_VERSION}+rke2r1
```

```suggestion
  replicas: ${CONTROL_PLANE_MACHINE_COUNT}
  version: ${KUBERNETES_VERSION}+rke2r1
```
After the RKE2 is bumped? Is this how it should look like?

---

yes but old version field will also be valid for some time

---

The fact that API was still in alpha was the only chance to avoid this “both are valid” behavior. This will be another problem to solve in the future